src/instana/instrumentation/pep0249.py

```diff
@@ -2,139 +2,169 @@
 # (c) Copyright Instana Inc. 2018
 
 # This is a wrapper for PEP-0249: Python Database API Specification v2.0
-import opentracing.ext.tags as ext
+from __future__ import annotations
 import wrapt
+from typing import TYPE_CHECKING
 
-from ..log import logger
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
-from ..util.sql import sql_sanitizer
+from opentelemetry.semconv.trace import SpanAttributes
+
+from instana.log import logger
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.util.sql import sql_sanitizer
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 
 class CursorWrapper(wrapt.ObjectProxy):
-    __slots__ = ('_module_name', '_connect_params', '_cursor_params')
+    __slots__ = ("_module_name", "_connect_params", "_cursor_params")
 
-    def __init__(self, cursor, module_name,
-                 connect_params=None, cursor_params=None):
+    def __init__(
+        self, cursor, module_name, connect_params=None, cursor_params=None
+    ) -> None:
         super(CursorWrapper, self).__init__(wrapped=cursor)
         self._module_name = module_name
         self._connect_params = connect_params
         self._cursor_params = cursor_params
 
-    def _collect_kvs(self, span, sql):
+    def _collect_kvs(self, span, sql) -> InstanaSpan:
         try:
-            span.set_tag(ext.SPAN_KIND, 'exit')
-
-            db_parameter_name = next((p for p in ('db', 'database', 'dbname') if p in self._connect_params[1]), None)
+            span.set_attribute("span.kind", "exit")
+
+            db_parameter_name = next(
+                (
+                    p
+                    for p in ("db", "database", "dbname")
+                    if p in self._connect_params[1]
+                ),
+                None,
+            )
             if db_parameter_name:
-                span.set_tag(ext.DATABASE_INSTANCE, self._connect_params[1][db_parameter_name])
-
-            span.set_tag(ext.DATABASE_STATEMENT, sql_sanitizer(sql))
-            span.set_tag(ext.DATABASE_USER, self._connect_params[1]['user'])
-            span.set_tag('host', self._connect_params[1]['host'])
-            span.set_tag('port', self._connect_params[1]['port'])
+                span.set_attribute(
+                    SpanAttributes.DB_NAME,
+                    self._connect_params[1][db_parameter_name],
+                )
+
+            span.set_attribute(SpanAttributes.DB_STATEMENT, sql_sanitizer(sql))
+            span.set_attribute(SpanAttributes.DB_USER, self._connect_params[1]["user"])
+            span.set_attribute("host", self._connect_params[1]["host"])
+            span.set_attribute("port", self._connect_params[1]["port"])
         except Exception as e:
             logger.debug(e)
         return span
 
-    def __enter__(self):
+    def __enter__(self) -> CursorWrapper:
         return self
 
-    def execute(self, sql, params=None):
+    def execute(self, sql, params=None) -> None:
```

Comment:
Missing type hints for the arguments here, and the return must be corrected.

Reply:
execute returns null

![image](https://github.com/user-attachments/assets/f47a92ce-2d76-4988-853d-6f33f65604c6)